### ISSUE TITLE
openssl external-test-pyca (1.51.0, 3.9) failures since yesterday

### ISSUE THREAD

**Issue by bernd-edlinger** [pyca/cryptography#10310]
I see suddenly lots of test failures, e.g. in https://github.com/bernd-edlinger/openssl/actions/runs/7715807456/job/21031191093

```
2024-01-30T18:56:00.8038385Z =================================== FAILURES ===================================
2024-01-30T18:56:00.8039006Z _____________________ test_from_encoded_point_not_a_curve ______________________
2024-01-30T18:56:00.8039926Z [gw3] linux -- Python 3.9.18 /home/runner/work/openssl/openssl/venv-cryptography/bin/python
2024-01-30T18:56:00.8040452Z 
2024-01-30T18:56:00.8040642Z     def test_from_encoded_point_not_a_curve():
2024-01-30T18:56:00.8041117Z         with pytest.raises(TypeError):
2024-01-30T18:56:00.8041709Z             with pytest.warns(CryptographyDeprecationWarning):
2024-01-30T18:56:00.8042306Z >               ec.EllipticCurvePublicNumbers.from_encoded_point(
2024-01-30T18:56:00.8042961Z                     "notacurve", b"\x04data"  # type: ignore[arg-type]
2024-01-30T18:56:00.8043500Z                 )
2024-01-30T18:56:00.8043735Z 
2024-01-30T18:56:00.8043895Z tests/hazmat/primitives/test_ec.py:226: 
2024-01-30T18:56:00.8044438Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2024-01-30T18:56:00.8044859Z 
2024-01-30T18:56:00.8045387Z cls = <class 'cryptography.hazmat.primitives.asymmetric.ec.EllipticCurvePublicNumbers'>
2024-01-30T18:56:00.8046103Z curve = 'notacurve', data = b'\x04data'
2024-01-30T18:56:00.8046413Z 
2024-01-30T18:56:00.8046533Z     @classmethod
2024-01-30T18:56:00.8046953Z     def from_encoded_point(
2024-01-30T18:56:00.8047379Z         cls, curve: EllipticCurve, data: bytes
2024-01-30T18:56:00.8047856Z     ) -> "EllipticCurvePublicNumbers":
2024-01-30T18:56:00.8048414Z         if not isinstance(curve, EllipticCurve):
2024-01-30T18:56:00.8048981Z >           raise TypeError("curve must be an EllipticCurve instance")
2024-01-30T18:56:00.8049563Z E           TypeError: curve must be an EllipticCurve instance
2024-01-30T18:56:00.8050009Z 
2024-01-30T18:56:00.8050594Z ../venv-cryptography/lib/python3.9/site-packages/cryptography/hazmat/primitives/asymmetric/ec.py:408: TypeError
2024-01-30T18:56:00.8051235Z 
2024-01-30T18:56:00.8051537Z During handling of the above exception, another exception occurred:
2024-01-30T18:56:00.8051930Z 
2024-01-30T18:56:00.8052375Z     def test_from_encoded_point_not_a_curve():
2024-01-30T18:56:00.8052833Z         with pytest.raises(TypeError):
2024-01-30T18:56:00.8053512Z             with pytest.warns(CryptographyDeprecationWarning):
2024-01-30T18:56:00.8054213Z >               ec.EllipticCurvePublicNumbers.from_encoded_point(
2024-01-30T18:56:00.8054844Z                     "notacurve", b"\x04data"  # type: ignore[arg-type]
2024-01-30T18:56:00.8055334Z                 )
2024-01-30T18:56:00.8056275Z E               Failed: DID NOT WARN. No warnings of type (<class 'cryptography.utils.CryptographyDeprecationWarning'>,) were emitted.
2024-01-30T18:56:00.8057052Z E                Emitted warnings: [].
2024-01-30T18:56:00.8057372Z 
2024-01-30T18:56:00.8057564Z tests/hazmat/primitives/test_ec.py:226: Failed
```

what does this mean?

**Comment by bernd-edlinger** [comment]
Hmm, in comparison to the last successful test result https://github.com/bernd-edlinger/openssl/actions/runs/7668538755/job/20900529989

The only difference I see is that previously pytest-7.4.4 was used and now the test engine picks pytest-8.0.0...

**Comment by alex** [comment]
I really couldn't tell you what this means -- that line of code has not been present in cryptography since 2022: 2b6e463c786e767a7ce50e88b2577148c4764a01

**Comment by h-vetinari** [comment]
> The only difference I see is that previously pytest-7.4.4 was used and now the test engine picks pytest-8.0.0...

pytest 8 broke a bunch of things, in particular by becoming stricter about (not) swallowing other warnings/errors that are (not) being raised within a given `with pytest.{error.warn}` context manager. So it's very likely that just capping the version there would help.

> I really couldn't tell you what this means -- that line of code has not been present in cryptography since 2022: https://github.com/pyca/cryptography/commit/2b6e463c786e767a7ce50e88b2577148c4764a01

The last [update](https://github.com/openssl/openssl/commit/a8e7bc7c7445c0323ca9a0a9c96cf5ce5e2c82f1) of the submodule is a while back, but interestingly, this went until https://github.com/pyca/cryptography/commit/7e33b0e7739d633c77b8c478620167f693ed13f4, which should have come _after_ 2b6e463c786e767a7ce50e88b2577148c4764a01.

Could it be that something's going wrong with how that submodule is built?

**Comment by bernd-edlinger** [comment]
Thanks for your replies.

I was aware that the openssl git repository references the pyca-cryptograpgy as a submodule but
I was surprised that it uses a fixed commit hash: fa84d185c0fefa7d612759e02a26def2f948fe34
in my case.  It looks like the issue goes away if I update that to a slightly newer commit.

**Comment by bernd-edlinger** [comment]
A fix was merged to the affected openssl branches.
Thus closing this.
Thanks!
